data/js/skill/craft/tailoring.js

```diff
@@ -1,79 +1,272 @@
-const itemDetailsScriptID = 4026;
-const craftGumpID = 4027;
-
+const textHue = 0x480;					// Color hue for all text in the crafting gump
+const scriptID = 4030;					// Script ID used to identify and close this gump
+const gumpDelay = 2000;					// Delay (ms) before gump reappears after crafting
+const repairDelay = 200;				// Delay (ms) before gump reappears after selecting a resource
+const itemDetailsScriptID = 4026;		// Script ID used to show item detail tooltips
+const craftGumpID = 4027;				// TriggerEvent ID used to build the crafting gump UI
+const itemsPerPage = 10;				// Number of craftable items shown per gump subpage
+const displayUnlearnedRecipes = true;	// Show recipes player hasn�t learned
+//////////////////////////////////////////////////////////////////////////////////////////
+// UOX3 Tailoring CraftingMap Configuration
+//
+// Description:
+// This file defines the `CraftingMap` used to populate the tailoring crafting gump in UOX3.
+// Each entry links a craftable item (defined in `create/tailoring.dfn`) to its display name,
+// gump page, and crafting behavior. This setup powers the dynamic UI for tailoring.
+//
+// Data Sources:
+// - `makeID` refers to the unique item ID defined in `create/tailoring.dfn`.
+// - `dictID` refers to a localized string entry defined in `dictionaries/dictionary.eng`.
+// - `customName` is an optional hardcoded string that overrides dictionary text if present.
+// - `recipeID` is an optional ID that locks the item behind a learned recipe.
+//
+// Entry Format:
+//   makeID: {
+//     dictID: <dictionaryID>,       // (Optional) Used for localized names via dictionary.eng
+//     customName: "<item name>",    // (Optional) Overrides dictionary name if present
+//     page: <pageNumber>,           // Gump page to display item on
+//     timerID: <timerID>,           // Timer used to re-show gump after crafting
+//     recipeID: <recipeID>          // (Optional) Requires the player to learn the recipe before crafting
+//   }
+//
+// Display Logic:
+// - If `customName` is present, it is used directly in the crafting gump.
+// - If `dictID` is present (and `customName` is not), `GetDictionaryEntry()` is used.
+// - If both are missing, the gump will show a fallback like `[Unnamed Item: ####]`.
+// - The `page` field determines which category tab the item appears under.
+// - If `recipeID` is present and the player has not learned it, the item will either be:
+//     - Hidden entirely (default behavior), or
+//     - Displayed but uncraftable, depending on `displayUnlearnedRecipes` setting.
+//
+// Example Entries:
+//
+// // Standard item using dictionary ID for localization
+// 130: { dictID: 11415, page: 1, timerID: 1 },
+//
+// // Custom item with hardcoded display name, no dictionary entry required
+// 999: { customName: "harrys sword", page: 1, timerID: 1 },
+//
+// // Recipe-locked item (only craftable if recipe is learned)
+// � 185: { dictID: 11469, page: 8, timerID: 8, recipeID: 185 },
```

Comment:
Be careful about copying in extra character symbols from external sources. There are several of these "diamond with question mark"-symbols in this script.